internal/controllers/clusterextension_controller.go

```diff
@@ -206,6 +207,10 @@ func (r *ClusterExtensionReconciler) reconcile(ctx context.Context, ext *ocv1.Cl
 	installedBundle, err := r.InstalledBundleGetter.GetInstalledBundle(ctx, ext)
 	if err != nil {
 		setInstallStatus(ext, nil)
+		var saerr *authentication.ServiceAccountNotFoundError
+		if errors.As(err, &saerr) {
+			err = saerr
```

Comment:
```suggestion
			setInstalledStatusConditionUnknown(ext, saerr.Error())
                         setStatusProgressing(ext, errors.New("installation cannot proceed due to missing ServiceAccount"))
                         return ctrl.Result{}, err
```

It seems misslanding.
We should provide the specific info for the status condition ( the new error ) 
and return directly.